app/models/page.rb

```diff
@@ -504,6 +505,41 @@ def formatted_plaintext_doc(doc)
     doc.text.sub(/^\s*/m, '').gsub(/ *$/m,'')
   end
 
+  def formatted_plaintext_table(table_element)
+    text_table = ""
+
+    # clean up in-cell line-breaks
+    table_element.xpath('//lb').each { |n| n.replace(' ')}
+
+    # calculate the widths of each column based on max(header, cell[0...end])
+    column_count = ([table_element.xpath("//th").count] + table_element.xpath('//tr').map{|e| e.xpath('td').count }).max
+    column_widths = {}
+    1.upto(column_count) do |column_index|
+      longest_cell = (table_element.xpath("//tr/td[position()=#{column_index}]").map{|e| e.text().length}.max || 0)
+      corresponding_heading = heading_length = table_element.xpath("//th[position()=#{column_index}]").first
+      heading_length = corresponding_heading.nil? ? 0 : corresponding_heading.text().length
+      column_widths[column_index] = [longest_cell, heading_length].max
+    end
+
+    # print the header as markdown
+    cell_strings = []
+    table_element.xpath("//th").each_with_index do |e,i|
+      cell_strings << e.text.rjust(column_widths[i+1], ' ')
+    end
+    text_table << cell_strings.join(' | ') << "\n"
+
+    # print the separator
+    text_table << column_count.times.map{|i| ''.rjust(column_widths[i+1], '-')}.join(' | ') << "\n"
+
+    # print each row as markdown
+    table_element.xpath('//tr').each do |row_element|
+      text_table << row_element.xpath('td').map{|e| e.text.rjust(column_widths[e.path.match(/.*(\d)/)[1].to_i], ' ') }.join(' | ') << "\n"
```

Comment:
If a spreadsheet project is sufficient blank/has nils of some sort then the plaintext export of that spreadsheet will fail on this line with a 
`no implicit conversion from nil to integer`

The row in my test data is:
`	
#<Nokogiri::XML::Element:0x306c4 name="tr" children=[#<Nokogiri::XML::Element:0x3055c name="td" children=[#<Nokogiri::XML::Text:0x30548 "Andreson John">]>, #<Nokogiri::XML::Element:0x30584 name="td" children=[#<Nokogiri::XML::Text:0x30570 "2">]>, #<Nokogiri::XML::Element:0x305ac name="td" children=[#<Nokogiri::XML::Text:0x30598 "W">]>, #<Nokogiri::XML::Element:0x305d4 name="td" children=[#<Nokogiri::XML::Text:0x305c0 "Railroad">]>, #<Nokogiri::XML::Element:0x305fc name="td" children=[#<Nokogiri::XML::Text:0x305e8 "836">]>, #<Nokogiri::XML::Element:0x30624 name="td" children=[#<Nokogiri::XML::Text:0x30610 "From Guard">]>, #<Nokogiri::XML::Element:0x30638 name="td">, #<Nokogiri::XML::Element:0x3064c name="td">, #<Nokogiri::XML::Element:0x30660 name="td">, #<Nokogiri::XML::Element:0x30534 name="td">, #<Nokogiri::XML::Element:0x30674 name="td">, #<Nokogiri::XML::Element:0x30688 name="td">, #<Nokogiri::XML::Element:0x306b0 name="td" children=[#<Nokogiri::XML::Text:0x3069c "false">]>]>`